src/lib/tsreader/TSReader.cpp

```diff
@@ -225,8 +235,8 @@ namespace MPTV
 
             if ( !m_rtspClient->OpenStream(m_fileName.c_str()) )
             {
-                SAFE_DELETE(m_rtspClient);
-                SAFE_DELETE(m_buffer);
+                SafeDelete(m_rtspClient);
+                SafeDelete(m_buffer);
                 return E_FAIL;

```

Comment:
You changed on the others to bool, only is in this class still `long` and e.g. `E_FAIL` used (where 1, the `S_OK` is 0).

Also on e.g. `long retval = m_fileReader->OpenFile(m_fileName);` becomes the boolean return checked with old.